### DISCUSSION TITLE
Install Wix4 and extensions in offline environment

### DISCUSSION THREAD

**Discussion by bmtpolo** [wixtoolset/issues#7515] [Questions]
I'm trying to install wix4 in an offline environment and having an issue.
I'm able to install Wix4 successfully with the following:
dotnet tool install -g -add-source {path} wix    # where path is local location of wix.4.0.0.0.nupkg file

However I'm not having any success getting the wix extensions (wixtoolset.bal.wixext, wixtoolset.difxapp.wixext, and wixtoolset.ui.wixext) to successfully install offline.  I do have the .nupkg for each of them.

Any help/suggestions would greatly be appreciated.

**Comment by barnson** [comment]
https://wixtoolset.org/docs/tools/wixext/#loading-extensions-using-the-wixexe-command-line-tool

**Reply by bmtpolo** [reply]
Thanks, I saw the documentation there but I'm not sure it answers the question.  I'm able to add the extension if I have an internet connection.  But if I have no internet connection I'm not able to get it to work.  For instance I downloaded the wixtoolset.util.wixext and copied it to my c:\temp directory.  Then I disabled my network connection and ran the command "wix extension add -g c:\temp\wixtoolset.util.wixext" and it errored out (I attached a file showing the results). 

[wix_add_extension_offline_error.txt](https://github.com/wixtoolset/issues/files/11569568/wix_add_extension_offline_error.txt)

My issue is I'm trying to setup an environment to build a wix application on a non-internet connected computer.

**Reply by barnson** [reply]
WiX uses the NuGet API so you'd need to configure NuGet to use a local cache or LAN-local NuGet server.

**Comment by jbohrmann** [comment]
For me adding the following PropertyGroup to my ".wixproj" did the job in an offline environment on a Windows Server:
```
<PropertyGroup>
  <RestoreSources>$(RestoreSources);D:/LocalPackageSource;</RestoreSources>
</PropertyGroup>
```
Then I put the WixToolset-nupkg-files to the directory LocalPackageSource;
```
D:\LocalPackageSource>dir /b
wixtoolset.heat.4.0.1.nupkg
wixtoolset.netfx.wixext.4.0.1.nupkg
wixtoolset.sdk.4.0.1.nupkg
wixtoolset.ui.wixext.4.0.1.nupkg
wixtoolset.util.wixext.4.0.1.nupkg
```
